### PR TITLE
React 6.3 -> Good Coding Practices

### LINKED ISSUE #6925: Frontend_with_Framework: React 6.3 -> Good Coding Practices
### Description

Contributors are required to create Documentation and explain about the Good Coding Practices which should cover the questions like:

- Why Good Coding practices is necessary?
- Benefits of good coding practices
- How it is helpful to a developer ?
- also give some theory explanation with code
- any addons from your side (in-depth of the topic) .

Note :

- Plagiarism is strictly prohibited.
- **Changes should be made inside the Frontend_Web_Development_React_Angular_Vue/ dir inside Frontend_Web_Development_React_Angular_Vue branch.**
- Files need to be added into the Week 6 folder only.

### Domain

Frontend Dev React/Angular/Vue

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by shreya024**
<hr>

## Description 📜

Added code and documentation for Good Coding Practices. 
Fixes #6925 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [x] Frontend Dev React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Frontend_Web_Development_React_Angular_Vue/React/Week 6/Good Coding Practices.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+- **Production-Quality Code** : Following good coding practices makes sure that code meets expectations for production in reliability,efficiency and other aspects.
+- **Clean Code** : Good Coding Practices make the code readable,simple and concise.
+- **Modular Code** : Modular production code makes the code more organized,efficient and reusable.
+- **Structure** : Refactoring ensures that there is improvement of internal structure without changing the external functionality of the code.
+- **Documentation** : Good Coding Practices advocate adding a relevant documentation. It is helpful for clarifying complex parts of the code and makes it easily understandable. Example: Adding a README file is a minimum requirement.
+- **Version Control** : It is always advisable to make meaningful commits after making changes and use version control tools like git. Merging commits to stay updated and resolving merge commits properly are integral part of following good coding practices.
+
+## Some Good Coding Practices in React
+
+- **Reusability** : Code Reusability can be schieved in React with the help of reusable components.
```
Comment (anandhu720) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 6/Good Coding Practices.md:23]: Spelling mistake

**Review by anandhu720 — CHANGES_REQUESTED** [review]

File: Frontend_Web_Development_React_Angular_Vue/React/Week 6/Good Coding Practices.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
+<h1 align="center">  Good Coding Practices </h1>
```
Comment (anandhu720) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 6/Good Coding Practices.md:1]: I think you can also add some details of folder structers also

**Review by anandhu720 — CHANGES_REQUESTED** [review]

**Comment by shreya024** [comment]
@anandhu720 Sure, I have made the suggested changes. Kindly check now.

**Review by heet-kumar — APPROVED** [review]
@shreya024  Nice PR LGTM! let's wait for the supervisor approval if they suggest any more changes then include that changes to.

**Comment by shreya024** [comment]
@heet-kumar Okay and thank you.

**Review by anandhu720 — APPROVED** [review]

**Pull request merged**